google-ads-examples/src/main/java/com/google/ads/googleads/examples/planning/ForecastReach.java

```diff
@@ -0,0 +1,357 @@
+    for (ReachForecast point : response.getReachCurve().getReachForecastsList()) {
+      System.out.printf("%s,\t", request.getCurrencyCode());
+      System.out.printf("%s,\t", point.getCostMicros());
+      System.out.printf("%s,\t", point.getForecast().getOnTargetReach());
+      System.out.printf("%s,\t", point.getForecast().getOnTargetImpressions());
+      System.out.printf("%s,\t", point.getForecast().getTotalReach());
+      System.out.printf("%s,\t", point.getForecast().getTotalImpressions());
+      System.out.print("\"[");
+      for (ProductAllocation product : point.getForecastedProductAllocationsList()) {
+        System.out.printf("Product: %s,", product.getPlannableProductCode());
+        System.out.printf("Budget Micros: %s", product.getBudgetMicros());
+      }
+      System.out.print("\"]%n");
+    }
+  }
+
+  /**
+   * Pulls a forecast for a budget split 15% and 85% between two products.
+   *
+   * @param reachPlanServiceClient instance of Reach Plan Service client.
+   * @param customerId the customer ID for the reach forecast.
+   * @param locationId location ID to plan for. To find a valid location ID, either see
+   *     https://developers.google.com/adwords/api/docs/appendix/geotargeting or call
+   *     ReachPlanServiceClient.listPlannableLocations().
+   * @param currencyCode three-character ISO 4217 currency code.
+   * @param budgetMicros budget in currency to plan for.
+   */
+  private void forecastManualMix(
+      ReachPlanServiceClient reachPlanServiceClient,
+      long customerId,
+      String locationId,
+      String currencyCode,
+      long budgetMicros) {
+    List<PlannedProduct> productMix = new ArrayList<>();
+
+    // Set up a ratio to split the budget between two products.
+    double trueviewAllocation = 0.15;
+    double bumperAllocation = 1 - trueviewAllocation;
+
+    // See listPlannableProducts on ReachPlanService to retrieve a list
+    // of valid PlannableProductCode's for a given location:
+    // https://developers.google.com/google-ads/api/reference/rpc/Google.Ads.GoogleAds.V4.services#reachplanservice
+    productMix.add(
+        PlannedProduct.newBuilder()
+            .setPlannableProductCode(StringValue.of("TRUEVIEW_IN_STREAM"))
+            .setBudgetMicros(Int64Value.of((long) ((double) budgetMicros * bumperAllocation)))
+            .build());
+    productMix.add(
+        PlannedProduct.newBuilder()
+            .setPlannableProductCode(StringValue.of("BUMPER"))
+            .setBudgetMicros(Int64Value.of((long) ((double) budgetMicros * bumperAllocation)))
+            .build());
+
+    GenerateReachForecastRequest request =
+        buildReachRequest(customerId, productMix, locationId, currencyCode);
+
+    pullReachCurve(reachPlanServiceClient, request);
+  }
+
+  /**
+   * Pulls a forecast for a product mix suggested based on preferences for whether the ad would have
+   * a guaranteed price, play with sound, would be skippable, would include top content, and a
```

Comment:
nit: last clause should have a verb as well to match other clauses, e.g., ", and would have a desired ad length".

Reply:
Fixed. @AnashOommen we should update the golden as well.